runtime/browser/ui/native_app_window_linux.h

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2015 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef XWALK_RUNTIME_BROWSER_UI_NATIVE_APP_WINDOW_LINUX_H_
+#define XWALK_RUNTIME_BROWSER_UI_NATIVE_APP_WINDOW_LINUX_H_
+
+#include <string>
+
+#include "ui/views/controls/button/label_button.h"
+#include "xwalk/runtime/browser/ui/native_app_window_views.h"
+
+namespace views {
+class WebView;
+class Textfield;
+class ProgressBar;
+}
+
+namespace xwalk {
+
+class NativeAppWindowLinux : public NativeAppWindowViews,
+                             public views::ButtonListener {
+ public:
+  class Controller {
+   public:
+    ~Controller() {}
+    virtual void Back() = 0;
```

Comment:
this is rather "BackButtonPressed", right? this is something coming from UI and it is not Linux-specific, IMO it would be better to move it to "NativeAppWindowDelegate" (and same for below methods)